common.mjs

```diff
@@ -272,6 +272,63 @@ export class ServerError extends Error
   }
 }
 
+export class HTTPError extends Error {
+  constructor(message, httpData) {
+    super(message);
+    this.status = httpData.status;
+    this.method = httpData.method;
+    this.url = httpData.url;
+    this.headers = httpData.headers;
+    this.body = httpData.body;
+  }
+}
+
+export async function sendRecvJSON2(url, method='GET', headers={}, body=null) {
+  return sendRecvHTTP2(url, method, headers, body, 'application/json', {
+    encode: JSON.stringify,
+    decode: JSON.parse
+  });
+}
+
+
+export async function sendRecvHTTP2(url, method='GET', headers={}, body=null,
+  contentType, codec) {
+    let options = {
+      method: method,
+      headers: {...headers}
```

Comment:
Why do you have headers and specifically content-type? This is confusing. I would prefer if you left it completely general and left it up to the caller or restrict it completely.